### DISCUSSION TITLE
[QUESTION] list of variables

### DISCUSSION THREAD

**Discussion by hasin89** [gijzelaerr/python-snap7#318] [Q&A] [answered]
Is there a way to list variables names from siemens s7-1200?
I don't even need the values, just the names of variables.

**Comment by spreeker** [comment, marked as answer]
I believe you have to get that from die siemens (TIA portal) / configuration software of siemens plcs. There is an export possibilities of all the addresses used by all variables and connected hardware.  Since any kind of configuration is possible in the PLC itself  won't contain easy to use variable names unless you put them in yourself.

**Comment by gijzelaerr** [comment]
Closing as answered. The S7 protocol does not provide variable/symbol names -- those must be exported from the TIA Portal project configuration.
